src/mem/ruby/protocol/GPU_VIPER-TCC.sm

```diff
@@ -345,7 +349,15 @@ machine(MachineType:TCC, "TCC Cache")
         DPRINTF(RubySlicc, "%s\n", in_msg);
         Entry cache_entry := getCacheEntry(in_msg.addr);
         TBE tbe := TBEs.lookup(in_msg.addr);
-        trigger(Event:PrbInv, in_msg.addr, cache_entry, tbe);
+        if (in_msg.Type == ProbeRequestType:PrbInv) {
+          // Cache is write-through. Invalidate data and send it
+          // downstream
+          trigger(Event:PrbInv, in_msg.addr, cache_entry, tbe);
+        } else {
+          // Cache is writeback. If data present in cache, then downgrade
+          // it and send it downstream
+          trigger(Event:PrbDowngrade, in_msg.addr, cache_entry, tbe);
+        }
```

Comment:
Could you explain this part? Do PrbInv requests only happen in write-through cache? I see elsewhere that `L2isWB` is used instead to check this, so does that imply PrbInv can happen in write-back cache too and the comment needs to be updated?

Reply:
I've updated the comments to remove the write-through/writeback part. It is more generic now. Thanks for pointing this out!